tests/core/test_model.py

```diff
@@ -1687,26 +1687,38 @@ def b_model(context):
 
     assert isinstance(python_model.kind, FullKind)
 
+    @model("kind_empty_dict", kind=dict(), columns={'"COL"': "int"})
+    def my_model(context):
+        pass
+
     # error if kind dict with no `name` key
     with pytest.raises(ConfigError, match="`kind` dictionary must contain a `name` key"):
+        python_model = model.get_registry()["kind_empty_dict"].model(
+            module_path=Path("."),
+            path=Path("."),
+        )
```

Comment:
Correct me if I'm wrong, but it looks like we haven't tested the name inference for python models. If that's true, let's add one.